service/header/p2p_exchange.go

```diff
@@ -13,10 +13,11 @@ import (
 
 	"github.com/celestiaorg/go-libp2p-messenger/serde"
 
+	params "github.com/celestiaorg/celestia-node/params"
 	pb "github.com/celestiaorg/celestia-node/service/header/pb"
 )
 
-var exchangeProtocolID = protocol.ID("/header-ex/v0.0.1")
+var exchangeProtocolID = protocol.ID(fmt.Sprintf("/header-ex-%s/v0.0.1", params.GetNetwork()))
```

Comment:
I believe the format should be like `/header-ex/v0.0.01/network`. Sorry for not mentioning this in the issue